sqlglot/dialects/redshift.py

```diff
@@ -388,13 +389,28 @@ def unnest_sql(self, expression: exp.Unnest) -> str:
             args = expression.expressions
             num_args = len(args)
 
-            if num_args > 1:
+            if not num_args == 1:
                 self.unsupported(f"Unsupported number of arguments in UNNEST: {num_args}")
                 return ""
 
-            arg = self.sql(seq_get(args, 0))
+            if not expression.find_ancestor(exp.From, exp.Join):
+                self.unsupported("Unsupported UNNEST when not used in FROM/JOIN clauses")
+                return ""
+
+            from sqlglot.optimizer.annotate_types import annotate_types
+
+            arg = t.cast(exp.Expression, seq_get(args, 0))
+            annotated_arg = annotate_types(arg)
+            if not annotated_arg.is_type(exp.DataType.Type.UNKNOWN) and not annotated_arg.is_type(
+                exp.DataType.Type.ARRAY
+            ):
+                self.unsupported("Unsupported UNNEST without ARRAY argument")
+                return ""
```

Comment:
I'm not sure about this bit, what triggered the need for this code path? Doesn't redshift have a `SUPER` type that can encode arrays too? Should we leave this out?

Reply:
I kept thinking that e.g. in DuckDB we can do `SELECT * FROM UNNEST(<struct expr>)` which will bypass the other checks (as the position of the `UNNEST` is valid in Redshift) but will not work properly due to the non-array expr.

However, the `SUPER` data type does make this more complicated than is needed, so will revert this change